### ISSUE TITLE
Announcement: I'm setting this project on ice for now

### ISSUE THREAD

**Issue by Lusito** [Lusito/forget-me-not#288]
Looking at all the unanswered issues, you might have guessed it:

I didn't have much time to spend on this project lately and working on it alone is very taxing. There are so many use-cases, which need to be considered and so many holes in the Webextension API, making it really really hard to give you all the features you want while keeping the UI and logic understandable to the average user.

While working on version 3, new features have come out for Firefox, which made the problem even worse and I was at a loss how to solve the issues properly. In addition to that, the amount of time I was able to spend on FMN has decreased a lot.

Btw: If you think other extensions solve these problems properly, I can tell you, that they either just pretend or they only have a limited subset of what forget-me-not does.

Mozilla is really slow with adding the webextension improvements necessary to write a good cleaning extension. They seem to focus on adding other new features or [questionable UI "improvements"](https://www.omgubuntu.co.uk/2021/02/try-firefox-proton-redesign-ubuntu).

Recently, Mozilla announced, that they would [remove background pages in favor of service workers](https://blog.mozilla.org/addons/2021/05/27/manifest-v3-update/).
A little background: Currently, FMN keeps a state of all open tabs, current settings, etc. in memory so that cleanup can happen quickly without issues.

With the change Mozilla is planning, however, I would have to store all of your tabs in the browsers permanent storage. While this is theoretically possible, this also means, that I would have to store urls on your harddisk, i.e. maybe exactly the stuff you don't want to store there in the first place. This would go against the whole idea of forget-me-not.

Even if I did decide to do this and give user enough information to decide what they want to do, this would be a huge rewrite of the entire code-base, for which I don't have the time right now. I couldn't even start yet, since they haven't implemented it completely yet.

Don't get me started on the new android version of firefox.

So where does this leave forget-me-not? Honestly I'm not sure:
- I can't do another rewrite of the extension with all features left intact completely alone. Too much work, too little time.
- I might be able to write a stripped-down version of forget-me-not with a limited feature-set, which works without saving urls to the harddisk. That would mean, that I'd have to write a new version almost entirely from scratch, which is not as much work as rewriting with all features, but still a lot of work.

For now I'm going to ice this project (i.e. stop working on it for a while) until I figure out how to continue with it.
The extension will continue to work until Mozilla completely disables the background pages. Not sure when this will happen.

I will try to find some time soon to at least add all the translation improvements that some of you have have written and release a new version with this.

**Comment by codl** [comment]
Thanks for all your work so far

Regarding writing the tablist to disk, Firefox is already doing that as part of Session Restore, it's what lets you recover your tabs after a crash. As far as I know, it cannot be disabled. So if you were to write the tablist to disk, you wouldn't be leaking any information that wasn't already there.

**Comment by ElhemEnohpi** [comment]
I'm really sorry to hear this! FMN is the best addon in its class, and one of the most important for me in terms of fighting against being tracked. Thanks for all the effort. I hope Mozilla will take seriously the potential loss of this great extension, and others in the same situation, and provide some more reasonable way of moving forward. Have you written to Mozilla and explained the difficulties? It might be helpful...

**Comment by ghost** [comment]
Damn, that's a shame. Thanks very much for a great addon.

I hope it doesn't seem too rude of me but perhaps this would be a good place to ask fellow FMN users if they have any suggestions for alternatives moving forward?

**Comment by B00ze64** [comment]
@Lusito You could complain and explain why you need background pages on bug 1578286.

**Comment by ghost** [comment]
> I hope it doesn't seem too rude of me but perhaps this would be a good place to ask fellow FMN users if they have any suggestions for alternatives moving forward?

FWIW, I'm probably using Cookie Auto-Delete. There's no way to import settings, but you can run it with auto-clean disabled, alongside FMN, so tht you can take your time to set it up before you cut over.

Still welcoming any other suggestions!

**Comment by sonbua** [comment]
Santo: Vielen dank fuer ein wunderbar Addon!

I'm feeling sad that this project is being on ice. I had a bi-weekly to-do on my todoist to check whether #23 is going to be released with v3 :-)

I'm giving it a try by building it on my local machine and running the test suite first. However it gives me some errors upon installing the packages `npm i`. When trying to manually upgrade the version in `package.json`, I can run `npm i` successfully, but the tests failed for me. I think it is the version upgrade caused the tests failing.

~~Can you list the prerequisite for the development environment of this addon, such as Node.js version, or any other necessary requirement?~~

**Update:** I upgraded major version of `mocha`, which caused test code fail to build (`MochaDone` was deprecated and removed from v8 onwards.) So I'm able to run all the tests now.

**Comment by ElhemEnohpi** [comment]
> I hope it doesn't seem too rude of me but perhaps this would be a good place to ask fellow FMN users if they have any suggestions for alternatives moving forward?

Yeah, Cookie Auto-Delete is probably the best alternative. It's missing some features compared to FMN though, for example the ability to have a "default deny" cookie policy, and temporary exception rules, among other things. These days though, with network partitioning (first-party isolation) it's much less necessary to block cookies for all websites by default. I don't do that anymore.

**Comment by MPeti1** [comment]
Well, I don't really think CAD is an alternative. The most important thing to me in this addon is that it can operate in blacklist mode (or whitelist mode if you want), as in only deleting what I say to delete. How CAD works is the opposite: you can only say what not to delete. And why is it that? Because the CAD author "believes" that blacklisting mode is worthless.

**Comment by iSigil** [comment]
This is disappointing.

**Comment by secretmango** [comment]
> Thanks for all your work so far
> 
> Regarding writing the tablist to disk, Firefox is already doing that as part of Session Restore, it's what lets you recover your tabs after a crash. As far as I know, it cannot be disabled. So if you were to write the tablist to disk, you wouldn't be leaking any information that wasn't already there.

Sessionrestore can be disabled, its part of the Arkenfox settings.

I would like to add that I only use this addon for cookies. Its great, and I guess it is extremely powerful. On Firefox Desktop there is the possibility to clear all cookies but exceptions, this is not there on mobile, so I use this Addon (Custom Addon Collection) and it works great!

If they maaaybe fix this on mobile too, it wouldnt be a problem anymore.

I guess I havent used the possible features at all, but maybe CookieAutoDelete is an option, yes.

Thanks a lot for your work!

**Comment by secretmango** [comment]
Just saying, on mobile at least the cookie deletion section still works perfectly
